File: gitdot-api/src/endpoint/organization/get_organization.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::organization::OrganizationResource};

pub struct GetOrganization;

impl Endpoint for GetOrganization {
    const PATH: &'static str = "/organization/{org_name}";
    const METHOD: http::Method = http::Method::GET;

    type Request = GetOrganizationRequest;
    type Response = GetOrganizationResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct GetOrganizationRequest {}

pub type GetOrganizationResponse = OrganizationResource;

File: gitdot-web/packages/gitdot-api/src/endpoint/organization/get-organization.ts
import { z } from "zod";
import { OrganizationResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const GetOrganizationRequest = z.object({});
export type GetOrganizationRequest = z.infer<typeof GetOrganizationRequest>;

export const GetOrganizationResponse = OrganizationResource;
export type GetOrganizationResponse = z.infer<typeof GetOrganizationResponse>;

export const GetOrganization = {
  path: "/organization/{org_name}",
  method: "GET",
  request: GetOrganizationRequest,
  response: GetOrganizationResponse,
} as const satisfies Endpoint;
export type GetOrganization = typeof GetOrganization;
